internal/checks/checks_test.go

```diff
@@ -141,3 +145,145 @@ func TestSleepCtx(t *testing.T) {
 	require.Error(t, err)
 	require.ErrorIs(t, err, context.Canceled)
 }
+
+// TestHandleCheckOp is testing internal functions that run as part of
+// updater.Run. Since these functions operate on scraper instances, a
+// test scraper is used, which in turn creates a test probe. The goal of
+// this is to decouple the testing of these functions from the testing
+// of the prober themselves.
+func TestHandleCheckOp(t *testing.T) {
+	publishCh := make(chan<- pusher.Payload, 100)
+
+	u, err := NewUpdater(
+		UpdaterOptions{
+			Conn:           new(grpc.ClientConn),
+			PromRegisterer: prometheus.NewPedanticRegistry(),
+			PublishCh:      publishCh,
+			TenantCh:       make(chan<- sm.Tenant),
+			Logger:         zerolog.Nop(),
+			ScraperFactory: testScraperFactory,
+		},
+	)
+
+	require.NotNil(t, u)
+	require.NoError(t, err)
+
+	u.probe = &sm.Probe{
+		Id:   100,
+		Name: "test-probe",
+	}
+
+	deadline, ok := t.Deadline()
+	if !ok {
+		deadline = time.Now().Add(2 * time.Second)
+	}
+
+	ctx, cancel := context.WithDeadline(context.Background(), deadline)
+	defer cancel()
+
+	check := sm.Check{
+		Id:        5000,
+		Frequency: 1000,
+		Timeout:   1000,
+		Target:    "127.0.0.1",
+		Job:       "", // not setting value to make check invalid
+		Probes:    []int64{1},
+		Settings: sm.CheckSettings{
+			Ping: &sm.PingSettings{},
+		},
+		Created:  0,
+		Modified: 0,
+	}
+
+	scraperExists := func() bool {
+		u.scrapersMutex.Lock()
+		defer u.scrapersMutex.Unlock()
+		_, found := u.scrapers[check.Id]
+		return found
+	}
+
+	// this should fail, check is invalid
+	err = u.handleCheckAdd(ctx, check)
+	require.Error(t, err)
+	// This doesn't work because the counter hasn't been set
+	// (because of the error):
+	// require.Equal(t, 0.0, testutil.ToFloat64(u.metrics.runningScrapers))
```

Comment:
err... if you read the code and follow the logic in the test, you'd expect the counter to be 0 at this point and _add_ the commented out line to verify that. The comment is explaining why that doesn't work, and "fixing" this (initializing the counters to zero, so that this doesn't fail) requires a non-trivial amount of changes.